src/loaders/ncbi_downloader/ncbi_downloader.py

```diff
@@ -34,18 +94,95 @@ def _download_genome_file(download_dir: str, gene_id: str, target_file_ext: list
                 for gene_file_name in gene_file_list:
                     if any([gene_file_name.endswith(ext) for ext in target_file_ext]):
                         result_file_path = os.path.join(download_dir, gene_file_name)
-                        host.download(gene_file_name, result_file_path)
+                        if overwrite or not os.path.exists(result_file_path):
+                            host.download(gene_file_name, result_file_path)
 
             success = True
         except Exception as e:
-            print(e)
+            print(f'Error:\n{e}\nfrom attempt {attempts + 1}.\nTrying to rerun.')
             attempts += 1
 
     if not success:
         raise ValueError(f'Download Failed for {gene_id} after {max_attempts} attempts!')
 
 
-def download_genome_files(gene_ids: list[str], target_file_ext: list[str], result_dir='ncbi_genomes') -> list[str]:
+def _form_gtdb_taxonomy_file_url(release_ver):
+    # form GTDB taxonomy URL for specific GTDB version.
+    # e.g. https://data.gtdb.ecogenomic.org/releases/release207/207.0/ar53_taxonomy_r207.tsv
+
+    gtdb_server = 'https://data.gtdb.ecogenomic.org/releases/'
+    file_dir_url = gtdb_server + f'release{release_ver}/' + f'{release_ver}.0/'
+
+    file_urls = list()
+    if release_ver in ['207', '202', '95', '89']:
```

Comment:
Hmm, ok. Depending on the page source is fragile at best so it seems we have two not-great choices (although the source is pretty simple). Depending on the page source means possibly less trouble in the future since it should work for new releases, but if the page source changes things might break (but that doesn't seem super likely). Doing it the way we're doing it now means less trouble now but a guarantee things'll break the next time a version is released. 

I'll leave it up to you, have fun

Reply:
👍  parsing GTDB web page for releasing versions now.

Reply:
:+1: